ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3MultipartUploadHandler.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2024 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ */
+package com.github.ambry.frontend.s3;
+
+import com.github.ambry.commons.Callback;
+import com.github.ambry.config.FrontendConfig;
+import com.github.ambry.frontend.AccountAndContainerInjector;
+import com.github.ambry.frontend.FrontendMetrics;
+import com.github.ambry.frontend.IdConverter;
+import com.github.ambry.frontend.NamedBlobPutHandler;
+import com.github.ambry.frontend.SecurityService;
+import com.github.ambry.named.NamedBlobDb;
+import com.github.ambry.quota.QuotaManager;
+import com.github.ambry.rest.RestMethod;
+import com.github.ambry.rest.RestRequest;
+import com.github.ambry.rest.RestResponseChannel;
+import com.github.ambry.rest.RestServiceErrorCode;
+import com.github.ambry.rest.RestServiceException;
+import com.github.ambry.router.ReadableStreamChannel;
+import com.github.ambry.router.Router;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.github.ambry.rest.RestUtils.*;
+
+
+/**
+ * Handles requests for s3 multipart uploads.
+ */
+public class S3MultipartUploadHandler {
```

Comment:
Good idea to have a class which handle the S3 request itself and it's separated from the http protocol like POST/PUT concept.